Makefile

```diff
@@ -2,15 +2,25 @@ ISTANBUL = node_modules/.bin/istanbul
 JSCS = node_modules/.bin/jscs
 JSHINT = node_modules/.bin/jshint
 NPM = npm
-XYZ = node_modules/.bin/xyz --repo git@github.com:plaid/sanctuary.git
+TRANSCRIBE = node_modules/.bin/transcribe
+XYZ = node_modules/.bin/xyz --repo git@github.com:plaid/sanctuary.git --script scripts/prepublish
 
-SRC = $(shell find . -name '*.js' -not -path './coverage/*' -not -path './node_modules/*')
+
+.PHONY: all
+all: README.md
+
+README.md: index.js
+	$(TRANSCRIBE) \
+	  --heading-level 4 \
+	  --url 'https://github.com/plaid/sanctuary/blob/v$(VERSION)/{filename}#L{line}' \
+	  -- $^ \
+	| sed 's/<h4 name="\(.*\)#\(.*\)">\(.*\)\1#\2/<h4 name="\1.prototype.\2">\3\1#\2/' >'$@'
```

Comment:
This replaces something like `name="Maybe#map"` with `name="Maybe.prototype.map"`. This is useful since according to [RFC 3986][1], `#` is not a valid character in a [fragment identifier][2]. I did test `https://github.com/…/README.md#Maybe#map` in Chrome, and it did work, but it seems wise to avoid unusual URLs which may not be handled correctly by any number of applications.


[1]: http://tools.ietf.org/html/rfc3986
[2]: http://tools.ietf.org/html/rfc3986#section-3.5